### PR TITLE
[Feature] Put.io

### PR DESCRIPTION
**Pull request by devclinton**
Fixes # feature http://feathub.com/SickRage/SickRage/+27

Proposed changes in this pull request:
- Add support for put.io client
- Updated code to user DEVELOP as base branch
- Changed URLs to Sickrage.io for client
- Change authentication to access token only and added a Get Token Link. There probably should be an update to https://sickrage.github.io to check for access_token in URL and show a div. 
- Fixed bug in Config Search form on verbiage update on Username/password fields

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked
- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

File: gui/slick/views/config_search.mako [hunk 1]
```diff
@@ -586,17 +586,19 @@
 
                                 <div class="field-pair" id="torrent_username_option">
                                     <label>
-                                        <span class="component-title" id="username_title">Client username</span>
+                                        <span class="component-title" id="torrent_username_title">Client username</span>
                                         <span class="component-desc">
                                             <input type="text" name="torrent_username" id="torrent_username" value="${sickbeard.TORRENT_USERNAME}" class="form-control input-sm input200" autocapitalize="off" autocomplete="no" />
-                                            <p>(blank for none)</p>
+                                            <p id="blank_username">(blank for none)</p>
+                                            <p id="get_token"><a href="https://api.put.io/v2/oauth2/authenticate?client_id=2392&response_type=token&redirect_uri=https://sickrage.github.io">Get Token</a>
```
Comment (OmgImAlexis) [inline gui/slick/views/config_search.mako:593]: This should have `&` encoded as `&amp;`.

**Comment by OmgImAlexis** [comment]
Does this take into account a directory so people can use `/tvshows` or does it just dump all files into the root directory in put.io?

**Comment by devclinton** [comment]
Since put.io is an online torrent service, you have to use an external
downloader. I have one on my github that can be used,
http://oudenmallon.github.io/putioDownloader/. I can enable the directory
option to be used with that if that is desired. Should I add a tutorial on
how to set that up somewhere?

**Comment by OmgImAlexis** [comment]
I know that, I'm talking about on put.io. You can add sub folders meaning if someone also uses this for CouchPotato as an example they can have that set to use /movies and this to use /tvshows meaning when the downloading software they use downloads/syncs the files they'll end up separated.

If this just adds them to the root put.io directory then they'll be all together with any other torrents they add.

@miigotu how do you feel about adding in put.io downloading like CouchPotato has?

**Comment by miigotu** [comment]
I think we need to worry about getting this part right first.  I kind of want to use the PR I made for it and maybe improve that, because I still see a few bugs in this one.

**Comment by devclinton** [comment]
I have no problems with that. I think it is an great idea. I should be able
to implement that last night. Should I also add an option for local folder
if the user has something to sync their put.io folder with their local
machine. I actually use my putioDownloader script in playlist mode so it
downloads a copy of all the files as a playlist instead of full files. This
allows me to have them indexes by Kodi.

**Comment by miigotu** [comment]
For the download sync I was thinking a resumable FTP client, since put.io has FTP. That would prevent broken downloads from restarting SR and it would also work for future planned FTP providers.

**Comment by devclinton** [comment]
FTP is a good idea. Would an FTP provider give options to manage the put.io
files there - IE delete files that have been already downloaded since you
have a finite space per account on put.io? Would it be possible to support
other download clients for it if it synced to a specific folder? I am
guessing the FTP sync would be part of Sickrage base so that would be a
better option overall but it would be nice to support others if possible.

**Comment by OmgImAlexis** [comment]
Put.io only provides a read only FTP server so deleting wouldn't work at all.

**Comment by miigotu** [comment]
You can use the API to delete though

**Comment by OmgImAlexis** [comment]
By the way for the downloading you can use aria2 like what CouchPotato does.

**Comment by miigotu** [comment]
#1550 @oudenmallon please work on it from that base.

**Comment by devclinton** [comment]
Will do. It may be Monday before I can work on it. I am leaving right after work today to the beach for the weekend. If I get a chance, I will try to test Saturday/Sunday morning. Worst case I will test after work on Monday. Thanks for the work @miigotu

**Comment by cvockrodt** [comment]
I personally can't get it working, not sure what client url to use in the options... 
I've tried (with and without ssl): 
https://api.put.io 
https://put.io 
https://sickrage.github.io
https://sickrage.io

but none of those seem to be working.

After some change, not sure what I did, there was a torrents option in the Manage menu, i.e. manage torrents, which brought up put.io in a frame inside sickrage. Thought that was weird.

Hope we can get some instruction on this

**Comment by miigotu** [comment]
It hasn't even bee n tested yet, as the tag log notes

**Comment by cvockrodt** [comment]
figured it'd still be in develop if it wasn't done, i'm seeing it in master.

**Comment by miigotu** [comment]
If you see how many fixes went out, it was important to release them. The put.io client causes no problems for the overall operation of the app if it doesnt work, so putting it out to get tested doesnt hurt, seeing as it requires an account just to even test it.

**Pull request closed without merging**